prefect_airbyte/connections.py

```diff
@@ -186,3 +189,186 @@ def example_trigger_sync_flow():
+            logger=self.airbyte_connection.logger,
+            timeout=self.airbyte_connection.timeout,
+        ) as airbyte_client:
+
+            job_status = JOB_STATUS_PENDING
+
+            while job_status not in terminal_job_statuses:
+                job_info = await airbyte_client.get_job_info(self.job_id)
+
+                job_status = job_info["job"]["status"]
+
+                # "recordsSynced" key is only present if the job has synced records
+                if "recordsSynced" in job_info["attempts"][-1]["attempt"]:
+                    self.records_synced = job_info["attempts"][-1]["attempt"][
+                        "recordsSynced"
+                    ]
+
+                # pending┃running┃failed┃succeeded┃cancelled
+                if job_status == JOB_STATUS_SUCCEEDED:
+                    self.logger.info(f"Job {self.job_id} succeeded.")
+                elif job_status in [JOB_STATUS_FAILED, JOB_STATUS_CANCELLED]:
+                    self.logger.error(f"Job {self.job_id} {job_status}.")
+                    raise err.AirbyteSyncJobFailed(f"Job {self.job_id} {job_status}.")
+                else:
+                    if self.airbyte_connection.status_updates:
+                        self.logger.info(job_status)
+                    # wait for next poll interval
+                    await sleep(self.airbyte_connection.poll_interval_s)
+
+    @sync_compatible
+    async def fetch_result(self) -> AirbyteSyncResult:
+        """Fetch the result of the `AirbyteSync`.
+
+        Returns:
+            `AirbyteSyncResult`: object containing metadata for the `AirbyteSync`.
+        """
+        async with self.airbyte_connection.airbyte_server.get_client(
+            logger=self.airbyte_connection.logger,
+            timeout=self.airbyte_connection.timeout,
+        ) as airbyte_client:
+            job_info = await airbyte_client.get_job_info(self.job_id)
+
+            job_status = job_info["job"]["status"]
+            job_created_at = job_info["job"]["createdAt"]
+            job_updated_at = job_info["job"]["updatedAt"]
+
+            return AirbyteSyncResult(
+                created_at=job_created_at,
+                job_id=self.job_id,
+                job_status=job_status,
+                records_synced=self.records_synced,
+                updated_at=job_updated_at,
+            )
+
+
+class AirbyteConnection(JobBlock):
+    """`JobBlock` representing an existing Airbyte connection.
+
+    Attributes:
+        airbyte_server: `AirbyteServer` block representing the Airbyte instance
+            where the Airbyte connection is defined.
+        connection_id: UUID of the Airbyte Connection to trigger.
+        poll_interval_s: Time in seconds between status checks of the Airbyte sync job.
+        status_updates: Whether to log job status on each poll of the Airbyte sync job.
+        timeout: Timeout in seconds for requests made by `httpx.AsyncClient`.
+
+    Example:
```

Comment:
Leaning towards a full example here. After loading, showing trigger, wait, fetch.
```suggestion
    Example:
```